packages/create-vite/src/index.ts

```diff
@@ -512,7 +512,12 @@ async function init() {
 
   pkg.name = packageName || getProjectName()
 
-  write('package.json', JSON.stringify(pkg, null, 2) + '\n')
+  if (template.endsWith('-ts')) {
+    if (!pkg.devDependencies) pkg.devDependencies = {}
+    pkg.devDependencies['@types/node'] = `^${process.versions.node}`
```

Comment:
process.versions.node may not match the version of @types/node published on npm

For example, odd numbers of versions are not available on npmjs, only even numbers of version are available.

Reply:
Thanks for the reminder. You're right, I overlooked this point. Moving forward, I will consider automatically adding @types/node only when the user's Node version meets Vite's requirements and is an even-numbered version. Otherwise, I'll output a prompt in the console. This might be a more refined solution.

Reply:
I think this still has issue. For example, at the time of writing this post, the latest node on https://nodejs.org/en is 22.11.0, but the latest @types/node on https://www.npmjs.com/package/@types/node?activeTab=versions is 22.9.0 . This will cause a mismatch and the package cannot be resolved.

Reply:
I guess we probably have to use npm's api to fetch all available versions and pick the most suitable version in the range?